libflux/flux/src/lib.rs

```diff
@@ -440,13 +437,14 @@ impl SemanticAnalyzer {
             return Err(fluxcore::Error::from(format!("{}", &errs[0])));
         }
 
-        let mut sem_pkg = fluxcore::semantic::convert::convert_with(ast_pkg, &mut self.f)?;
+        let mut fresher = fresher();
```

Comment:
This is the weird situation where this code needs to stay the way that it is. This code is special for the REPL and, unlike everywhere else, is reentrant. It's meant to work when a user inputs multiple commands into the REPL. As the REPL is essentially like gradually parsing a single main package, the fresher has to be maintained across invocations.

Reply:
True, as the `Environment` persists here, the fresher must also persist. However if applications on `PolyType` took shadowing into account then we wouldn't need to worry about variables being reused.

Optimally, we should only need to persist a `Fresher` throughout a set of mutually recursive recursive bindings. Outside of that all variables should be bound (is that the right term?).